src/app/api/handle-chapter-request/route.ts

```diff
@@ -0,0 +1,86 @@
+import { NextRequest, NextResponse } from "next/server";
+import {
+  HandleChapterRequest,
+  HandleChapterRequestResponse,
+} from "./route.schema";
+import { prisma } from "@server/db/client";
+
+export const POST = async (request: NextRequest) => {
+  // Validate the data in the request
+  // If the data is invalid, return a 400 response
+  // with a JSON body containing the validation errors
+
+  // Validate a proper JSON was passed in as well
+  try {
+    const handleChapterRequest = HandleChapterRequest.safeParse(
+      await request.json()
+    );
+    if (!handleChapterRequest.success) {
+      return NextResponse.json(
+        HandleChapterRequestResponse.parse({
+          code: "INVALID_REQUEST",
+          message: "Invalid API request",
+        }),
+        { status: 400 }
+      );
+    } else {
+      const body = handleChapterRequest.data;
+      // Check if the chapter request even exists
+      const chapterRequest = await prisma.chapterRequest.findFirst({
+        where: {
+          id: String(body.chapterRequestId),
+        },
+      });
+
+      if (!chapterRequest) {
+        return NextResponse.json(
+          HandleChapterRequestResponse.parse({
+            code: "CHAPTER_REQUEST_NOT_FOUND",
+            message:
+              "A chapter request associated with the given ID does not exist",
+          }),
+          { status: 400 }
+        );
+      }
+
+      // Regardless of approved/denied, delete the chapter request
+      await prisma.chapterRequest.delete({
```

Comment:
I think we don't want to delete the information. The client would like to display the information related to chapter requests. Can you guys add a new field to `ChapterRequest` document to mark it? Apologies for not making it clear.